pkg/operator/controllers/guardrails/gktemplates/aro-deny-delete.yaml

```diff
@@ -0,0 +1,31 @@
+apiVersion: templates.gatekeeper.sh/v1beta1
+kind: ConstraintTemplate
+metadata:
+  name: aro-deny-delete
+  annotations:
+    description: >-
+      Prevents unexpected deletion of a specific critical resource
+spec:
+  crd:
+    spec:
+      names:
+        kind: ARODenyDelete
+      validation:
+        openAPIV3Schema:
+          type: object
+          properties:
+            name:
+              type: string
+              description: >-
+                The specific resource name that is not allowed to be deleted.
+  targets:
+  - target: admission.k8s.gatekeeper.sh
+    rego: |
+      package arodenydelete
+
+      deny[msg] {
+        input.review.operation == "DELETE"
+        deny_resource_name := input.parameters.name
+        deny_resource_name == input.review.object.metadata.name
+        msg = "Deleting resource - " + name + " is not allowed"
+      }
```

Comment:
got some error messages like:
invalid ConstraintTemplate: invalid rego: invalid module: missing required rules: [violation] 

try modified some places and finally get it work:
```
    rego: |
      package arodenydelete
      violation[{"msg": msg}] {
        input.review.operation == "DELETE"
        deny_resource_name := input.parameters.name
        deny_resource_name == input.review.object.metadata.name
        msg := sprintf("Deleting resource - %v is not allowed", [deny_resource_name])
      }
```
`$ oc delete secrets pull-secret  -n openshift-config
Error from server (Forbidden): admission webhook "validation.gatekeeper.sh" denied the request: [aro-pull-secret-deny-delete] Deleting resource - pull-secret is not allowed
`

Reply:
thanks for the correction 👍🏼